python/xorbits/_mars/services/task/task_info_collector.py

```diff
@@ -299,36 +328,50 @@ class TaskInfoCollectorActor(mo.Actor):
     def __init__(self, profiling_config: Optional[Dict[str, Any]] = None):
         if profiling_config is None:
             profiling_config = dict()
-        self.profiling_config = profiling_config
-        self.experimental_profiling_config = profiling_config.get(
-            "experimental", dict()
-        )
-        mars_temp_dir = os.environ.get(MARS_LOG_PATH_KEY)
-        if mars_temp_dir is None:
-            self.yaml_root_dir = os.path.join(tempfile.tempdir, "mars_task_infos")
-        else:
-            self.yaml_root_dir = os.path.abspath(
-                os.path.join(mars_temp_dir, "../..", "mars_task_infos")
-            )
-        logger.info(f"Save task info in {self.yaml_root_dir}")
 
-    async def collect_task_info_enabled(self):
-        return self.experimental_profiling_config.get(
+        experimental_profiling_config = profiling_config.get("experimental", dict())
+        self._collect_task_info_enabled = experimental_profiling_config.get(
             "collect_task_info_enabled", False
         )
+        self._task_info_root_path = experimental_profiling_config.get(
+            "task_info_root_path", None
+        )
+        self._task_info_storage_options = experimental_profiling_config.get(
+            "task_info_storage_options", {}
+        )
+        if self._task_info_root_path is None:
+            profiling_results_dir = os.environ.get(MARS_PROFILING_RESULTS_PATH_KEY)
+            if profiling_results_dir is None:
+                profiling_results_dir = tempfile.mkdtemp(prefix=MARS_TMP_DIR_PREFIX)
```

Comment:
Do we need to manage these temporary paths? log and profiling use different paths.

Reply:
True. How about /tmp/mars_profiling ?

Reply:
I'm thinking to put them under the same directory:
- /tmp/mars
   - mars_log
   - mars_profiling

Reply:
After discussion, we agreed that the temp dir root path should be /tmp/mars on macOS / Linux. The logs should be in /tmp/mars/log and profiling results should be in /tmp/mars/profiling